gatewayconfig/bluetooth/characteristics/diagnostics_characteristic.py

```diff
@@ -25,19 +26,58 @@ def __init__(self, service, eth0_mac_address, wlan0_mac_address):
                 ["read"], service)
         self.add_descriptor(DiagnosticsDescriptor(self))
         self.add_descriptor(OpaqueStructureDescriptor(self))
-        self.p2pstatus = False
-        self.eth0_mac_address = eth0_mac_address
-        self.wlan0_mac_address = wlan0_mac_address
+
+        firmware_version = os.getenv('FIRMWARE_VERSION')
+        self.new_diagnostics_proto(eth0_mac_address, wlan0_mac_address,
+                                   firmware_version)
 
     def ReadValue(self, options):
         logger.debug('Read diagnostics')
         try:
-            self.p2pstatus = self.get_p2pstatus()
-            diagnostics = self.build_diagnostics_proto().SerializeToString()
-            logger.debug("Diagnostics are %s" % diagnostics)
-            return string_to_dbus_byte_array(diagnostics)
-        except Exception:
-            logger.exception("Unexpected exception while trying to read diagnostics")
+            p2pstatus = self.get_p2pstatus()
+
+            # Update self.diagnostics_proto
+            self.update_diagnostics_proto(p2pstatus)
+        except Exception as ex:
+            logger.exception("Unexpected exception while trying to read diagnostics: %s" % str(ex))
+
+        diagnostics = self.diagnostics_proto.SerializeToString()
+        logger.debug("Diagnostics are %s" % diagnostics)
+        return string_to_dbus_byte_array(diagnostics)
+
+    # Make a new diagnostics_pb2.diagnostics_v1
+    def new_diagnostics_proto(self,
+                              eth0_mac_address,
+                              wlan0_mac_address,
+                              firmware_version):
+        logger.debug('New Diagnostics Proto')
+
+        self.diagnostics_proto = diagnostics_pb2.diagnostics_v1()
+        self.diagnostics_proto.diagnostics['connected'] = DBUS_UNAVAILABLE_VALUE
+        self.diagnostics_proto.diagnostics['dialable'] = DBUS_UNAVAILABLE_VALUE
+        self.diagnostics_proto.diagnostics['height'] = DBUS_UNAVAILABLE_VALUE
+        self.diagnostics_proto.diagnostics['nat_type'] = DBUS_UNAVAILABLE_VALUE
+
+        self.diagnostics_proto.diagnostics['eth'] = eth0_mac_address.replace(":", "")
+        self.diagnostics_proto.diagnostics['wifi'] = wlan0_mac_address.replace(":", "")
+        self.diagnostics_proto.diagnostics['fw'] = str(firmware_version)
+        self.diagnostics_proto.diagnostics['ip'] = ""
```

Comment:
nit: this should probably be the same default in both places https://github.com/NebraLtd/hm-config/pull/136/files#diff-8cb7c89f6dc236110ee7edb3865ce64db989a52833158107310a2e1c3a5fc663L98

Reply:
`ip` default value still inconsistent. Here we set to empty string. But if we later can't update, the value falls back to `0.0.0.0` https://github.com/NebraLtd/hm-config/pull/136/files#diff-8cb7c89f6dc236110ee7edb3865ce64db989a52833158107310a2e1c3a5fc663L98

Reply:
> @posterzh are you still working on adding new unit tests?

@marvinmarnold yes

Reply:
I agree to set the initial value of ip with "0.0.0.0".

Reply:
I don't love using '0.0.0.0' as the default IP. I rather have it set to `None` as it is confusing to pass on '0.0.0.0' to the user.

Reply:
I will check IP address with `None` value doesn't make an exception right now.

Reply:
It's impossible to set `None` as default IP address because of the TypeError in `diagnostics_pb2.diagnostics_v1()`.
I would like to set "" as default IP.

Reply:
"" is better than 0.0.0.0 IMHO